frontend/src/views/CellGuide/components/CellGuideCard/connect.ts

```diff
@@ -0,0 +1,150 @@
+import { DefaultDropdownMenuOption } from "@czi-sds/components";
+import { throttle } from "lodash";
+import { useRouter } from "next/router";
+import { useState, useRef, useCallback, useMemo, useEffect } from "react";
+import {
+  ALL_TISSUES,
+  NO_ORGAN_ID,
+  TISSUE_AGNOSTIC,
+} from "src/views/CellGuide/components/CellGuideCard/components/MarkerGeneTables/constants";
+import { useOrganAndOrganismFilterListForCellType } from "src/views/CellGuide/components/CellGuideCard/components/MarkerGeneTables/hooks/common";
+import { SKINNY_MODE_BREAKPOINT_WIDTH } from "src/views/CellGuide/components/CellGuideCard/constants";
+import { LEFT_RIGHT_PADDING_PX_XXL } from "src/views/CellGuide/components/CellGuideCard/style";
+import { SDSOrgan } from "src/views/CellGuide/components/CellGuideCard/types";
+import { CellType } from "src/views/CellGuide/components/common/OntologyDagView/common/types";
+import { Gene } from "src/views/WheresMyGeneV2/common/types";
+
+export function useConnect() {
+  const router = useRouter();
+
+  const [pageNavIsOpen, setPageNavIsOpen] = useState(false);
+  const [selectedGene, setSelectedGene] = useState<string | undefined>(
+    undefined
+  );
+
+  const [skinnyMode, setSkinnyMode] = useState<boolean>(false);
+
+  // Navigation
+  const sectionRef0 = useRef<HTMLDivElement>(null);
+  const sectionRef1 = useRef<HTMLDivElement>(null);
+  const sectionRef2 = useRef<HTMLDivElement>(null);
+  const sectionRef3 = useRef<HTMLDivElement>(null);
+
+  const selectGene = useCallback(
+    (gene: string) => {
+      if (gene === selectedGene) {
+        setSelectedGene(undefined);
+      } else {
+        setSelectedGene(gene);
+        if (sectionRef1.current) {
+          window.scrollTo({
+            top:
+              sectionRef1.current.getBoundingClientRect().top +
+              window.scrollY -
+              50,
+            behavior: "smooth",
+          });
+        }
+      }
+    },
+    [selectedGene]
+  );
+
+  // Set the mobile tooltip view content
+  const [tooltipContent, setTooltipContent] = useState<{
+    title: string;
+    element: JSX.Element;
+  } | null>(null);
+
+  const { cellTypeId: queryCellTypeId, tissueId: queryTissueId } = router.query;
```

Comment:
since we get `tissueId` from the router now, I've removed the states that used to track `selectedOrgan` and `selectedOrganId`